From 0aa2e2cae2e1777863993a10c809bb50f4cde7f8 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Sat, 25 Apr 2020 20:55:28 +0200
Subject: [PATCH] fix boolean value with json-c 0.14

Upstream removed the TRUE and FALSE defines in commit
0992aac61f8b087efd7094e9ac2b84fa9c040fcd.
---
 ykpers-json.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

--- a/ykpers-json.c
+++ b/ykpers-json.c
@@ -40,7 +40,7 @@
 #define yk_json_object_object_get(obj, key, value) json_object_object_get_ex(obj, key, &value)
 #else
 typedef int json_bool;
-#define yk_json_object_object_get(obj, key, value) (value = json_object_object_get(obj, key)) == NULL ? (json_bool)FALSE : (json_bool)TRUE
+#define yk_json_object_object_get(obj, key, value) (value = json_object_object_get(obj, key)) == NULL ? 0 : 1
 #endif
 
 static void set_json_value(struct map_st *p, int mode, json_object *options, YKP_CONFIG *cfg) {
@@ -50,7 +50,7 @@ static void set_json_value(struct map_st
 	if(p->mode && (mode & p->mode) == mode) {
 		json_object *joption;
 		json_bool ret = yk_json_object_object_get(options, p->json_text, joption);
-		if(ret == TRUE && json_object_get_type(joption) == json_type_boolean) {
+		if(ret == 1 && json_object_get_type(joption) == json_type_boolean) {
 			int value = json_object_get_boolean(joption);
 			if(value == 1) {
 				p->setter(cfg, true);
@@ -230,20 +230,20 @@ int _ykp_json_import_cfg(YKP_CONFIG *cfg
 			ykp_errno = YKP_EINVAL;
 			goto out;
 		}
-		if(yk_json_object_object_get(jobj, "yubiProdConfig", yprod_json) == FALSE) {
+		if(yk_json_object_object_get(jobj, "yubiProdConfig", yprod_json) == 0) {
 			ykp_errno = YKP_EINVAL;
 			goto out;
 		}
-		if(yk_json_object_object_get(yprod_json, "mode", jmode) == FALSE) {
+		if(yk_json_object_object_get(yprod_json, "mode", jmode) == 0) {
 			ykp_errno = YKP_EINVAL;
 			goto out;
 		}
-		if(yk_json_object_object_get(yprod_json, "options", options) == FALSE) {
+		if(yk_json_object_object_get(yprod_json, "options", options) == 0) {
 			ykp_errno = YKP_EINVAL;
 			goto out;
 		}
 
-		if(yk_json_object_object_get(yprod_json, "targetConfig", jtarget) == TRUE) {
+		if(yk_json_object_object_get(yprod_json, "targetConfig", jtarget) == 1) {
 			int target_config = json_object_get_int(jtarget);
 			int command;
 			if(target_config == 1) {
@@ -275,13 +275,13 @@ int _ykp_json_import_cfg(YKP_CONFIG *cfg
 		if(mode == MODE_OATH_HOTP) {
 			json_object *jdigits, *jrandom;
 			ykp_set_tktflag_OATH_HOTP(cfg, true);
-			if(yk_json_object_object_get(options, "oathDigits", jdigits) == TRUE) {
+			if(yk_json_object_object_get(options, "oathDigits", jdigits) == 1) {
 				int digits = json_object_get_int(jdigits);
 				if(digits == 8) {
 					ykp_set_cfgflag_OATH_HOTP8(cfg, true);
 				}
 			}
-			if(yk_json_object_object_get(options, "randomSeed", jrandom) == TRUE) {
+			if(yk_json_object_object_get(options, "randomSeed", jrandom) == 1) {
 				int random = json_object_get_boolean(jrandom);
 				int seed = 0;
 				if(random == 1) {
@@ -290,7 +290,7 @@ int _ykp_json_import_cfg(YKP_CONFIG *cfg
 					goto out;
 				} else {
 					json_object *jseed;
-					if(yk_json_object_object_get(options, "fixedSeedvalue", jseed) == TRUE) {
+					if(yk_json_object_object_get(options, "fixedSeedvalue", jseed) == 1) {
 						seed = json_object_get_int(jseed);
 					}
 				}
